src/System.Data.SqlClient/tests/ManualTests/SQL/MirroringTest/ConnectionOnMirroringTest.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.Data.SqlClient.ManualTesting.Tests
+{
+    public static class ConnectionOnMirroringTest
+    {
+        [CheckConnStrSetupFact]
+        public static void TestMultipleConnectionToMirroredServer()
+        {
+            string connectionString = DataTestUtility.TcpConnStr;
+            int mirroringState;
+            string failoverPartnerName;
+            bool isMirroring = GetMirroringInfo(connectionString, out mirroringState, out failoverPartnerName);
+            List<SqlConnection> list = new List<SqlConnection>();
+            if (isMirroring && mirroringState == 4 && !string.IsNullOrEmpty(failoverPartnerName))
+            {
+                Stopwatch stopWatch = new Stopwatch();
+                TestWorker worker = new TestWorker(connectionString);
+                Thread childThread = new Thread(() => worker.TestMultipleConnection());
+
+                stopWatch.Start();
+                childThread.Start();
+                while (!worker.IsDone && stopWatch.ElapsedMilliseconds <= 10000);
+                stopWatch.Stop();
+
+                if (worker.IsDone)
+                {
+                    childThread.Join();
+                }
+                else
+                {
+                    //thread.Abort() is not implemented yet in CoreFx.
+                    childThread.Interrupt();
+                    throw new Exception();
+                }
+            }
+        }
+
+        private static bool GetMirroringInfo(string connectionString, out int mirroringState, out string failoverPartnerName)
+        {
+            mirroringState = -1;
+            failoverPartnerName = null;
+
+            SqlConnectionStringBuilder existingConnStrBuilder = new SqlConnectionStringBuilder(connectionString);
+            SqlConnectionStringBuilder newConnStrBuilder = new SqlConnectionStringBuilder();
+            newConnStrBuilder.DataSource = existingConnStrBuilder.DataSource;
+            if (!string.IsNullOrEmpty(existingConnStrBuilder.UserID))
```

Comment:
Why do you need to manually check all the state if you just copy the same values anyway?

Reply:
How do you clear the unwanted options out?

Reply:
https://msdn.microsoft.com/en-us/library/system.data.sqlclient.sqlconnectionstringbuilder.remove(v=vs.110).aspx